tests/surveytests.ts

```diff
@@ -12647,6 +12647,55 @@ QUnit.test(
     SurveyElement.FocusElement = oldFunc;
   }
 );
+QUnit.test(
+  "Focus question on Survey onServerValidateQuestions event, Bug#3343",
+  function (assert) {
+    var focusedQuestionId = "";
+    var oldFunc = SurveyElement.FocusElement;
+    SurveyElement.FocusElement = function (elId: string): boolean {
+      focusedQuestionId = elId;
+      return true;
+    };
+    var returnResult: (res: any) => void;
+    function asyncFunc(params: any): any {
```

Comment:
The function `asyncFunc` looks sync. Should we use `setTimeout` here?

Reply:
I use callback function to emulate async in sync unit test